### ISSUE TITLE
Breaking on printWidth when using class annotations

### ISSUE THREAD

**Issue by gcazaciuc** [jlongster/prettier#147]
Giving a code like
```js
class MyContractSelectionWidget extends React.Component<void,  MyContractSelectionWidgetPropsType, void> {
```
The printer puts all the code on a single line( printWidth limit is configured properly) leading to ESLint complaining that the line is too long.

**Comment by vjeux** [comment]
I can indeed repro:
https://jlongster.github.io/prettier/#%7B%22content%22%3A%22class%20MyContractSelectionWidget%20extends%20React.Component%3Cvoid%2C%20%20MyContractSelectionWidgetPropsType%2C%20void%3E%20%7B%5Cn%7D%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3Afalse%2C%22bracketSpacing%22%3Atrue%7D%7D

**Comment by vjeux** [comment]
Thanks, a fix is implemented in #466

**Comment by jlongster** [comment]
> The printer puts all the code on a single line( printWidth limit is configured properly) leading to ESLint complaining that the line is too long.

FWIW the maximum line width is a suggestion and prettier will attempt to wrap code so that it fits within it, but there are times when it can't wrap anymore and it lets code run past that limit. The beauty of this is when that happens, instead of having to mark that line with "// es-lint-ignore max-len" or whatever (making the line *more* ugly), you just don't have to worry about it. I'd recommend turning off the max-len rule.
